File: hospital_sys/app/Models/Department.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Department extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'location',
    ];

    // Relationships
    public function staff()
    {
        return $this->hasMany(Staff::class);
    }

    public function rooms()
    {
        return $this->hasMany(Room::class);
    }
}

File: app/models/department.py
# app/models/department.py
from app.db import db

class Department(db.Model):
    __tablename__ = "departments"

    department_id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String, nullable=False)
    location = db.Column(db.String)

    # Relationships
    staff_members = db.relationship("Staff", back_populates="department")
    rooms = db.relationship("Room", back_populates="department")
